lib/push.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var request = require('request')
+
+var ERR_NO_PUSH_CALLBACK = 'No Push Callback'
+var LOG_UPDATE_INCREMENT_SEND = 'push.send'
+var LOG_UPDATE_INCREMENT_SUCCESS = 'push.success'
+var LOG_UPDATE_INCREMENT_FAIL = 'push.failed'
+var LOG_UPDATE_INCREMENT_NO_CALLBACK = 'push.no_push_callback'
+
+module.exports = function (log, db) {
+  return {
+    /**
+     *  Notifies all devices that there was an update to the account
+     *
+     * @param uid
+     * @promise
+     */
+    notifyUpdate: function notifyUpdate(uid) {
+      var LOG_NOTIFY = 'push.notifyUpdate'
+
+      return db.devices(uid).then(
+        function (devices) {
+          log.trace({op: LOG_NOTIFY, devices: devices})
+
+          devices.forEach(function (device) {
+            if (device.pushCallback) {
+              log.trace({op: LOG_NOTIFY, pushCallback: device.pushCallback})
+
+              // send the push notification
+              log.increment(LOG_UPDATE_INCREMENT_SEND)
+              request.post(device.pushCallback, function (err){
+                if (err) {
+                  log.error({op: LOG_NOTIFY, err: err})
+                  log.increment(LOG_UPDATE_INCREMENT_FAIL)
+                } else {
+                  log.increment(LOG_UPDATE_INCREMENT_SUCCESS)
+                }
+              })
```

Comment:
I like that the API exposed by this module is fire-and-forget.  Two questions about the request handling here:

* Should we try to impose some parallelism limit here, in case the user has large number of devices?  Seems unlikely, but worth considering.
* Does the push spec require us to take an action in response to particular error codes, e.g. invalidating a callback URL if we get a `410 Gone`?